### PR TITLE
CLI tests with gecko node and speculos

### PR DESCRIPTION
**Pull request by jmininger**

### CODE REVIEW

File: cli-tests.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ cli-app-avalanche, gecko, pkgs ? import <nixpkgs> {}}:
+let
+  cli = cli-app-avalanche;
+  fetchThunk = p:
```
Comment (3noch) [inline cli-tests.nix:4]: We should pull in r-p for this. Multiple conversations have been had about this and that's what we've landed on for now.

File: cli-tests.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    if builtins.pathExists (p + /git.json)
+      then pkgs.fetchgit { inherit (builtins.fromJSON (builtins.readFile (p + /git.json))) url rev sha256; }
+    else if builtins.pathExists (p + /github.json)
+      then pkgs.fetchFromGitHub { inherit (builtins.fromJSON (builtins.readFile (p + /github.json))) owner repo rev sha256; }
+    else p;
+  speculos = pkgs.callPackage ./nix/dep/speculos { };
+  test-certs-dir = pkgs.copyPathToStore ./testnet/certs;
+
+  createEnvScript = name: env:
+		let setEnvVars = with pkgs.lib; concatStringsSep " && " (attrValues (mapAttrs (k: v: "${k}=${v}") env));
```
Comment (3noch) [inline cli-tests.nix:14]: Please no tabs.

File: cli/cli.js [hunk 3]
```diff
@@ -111,7 +111,7 @@ async function get_transport_with_wallet(devices, open, chosen_device, wallet_id
 async function makeWithTransport(options) {
   const [open, found_device] = options.speculos === undefined
     ? [TransportNodeHid.open, await get_transport_with_wallet(await TransportNodeHid.list(), TransportNodeHid.open, options.device, options.wallet)]
-    : [TransportSpeculos.open, await get_transport_with_wallet([{ apduPort: options.speculos }], TransportSpeculos.open, { apduPort: options.speculos } , options.wallet)];
+    : [TransportSpeculos.open, await get_transport_with_wallet([{ apduPort: options.speculos }], TransportSpeculos.open, { buttonPort: 8888, apduPort: options.speculos, automationPort:8899 } , options.wallet)];
```
Comment (3noch) [inline cli/cli.js:114]: These should be global constants at the very least, ideally env vars or something configurable.

File: cli/cli.js [hunk 4]
```diff
@@ -183,7 +183,12 @@ program
       path = AVA_BIP32_PREFIX + "/" + path;
       console.error("Getting public key for path", path);
       requestLedgerAccept();
+
+      if(options.speculos) {
```
Comment (3noch) [inline cli/cli.js:187]: Let's stick with consistent spacing here: `if (`
```suggestion
      if (options.speculos) {
```

File: cli/cli.js [hunk 5]
```diff
@@ -361,6 +366,10 @@ program
 
     if (address === undefined) {
       await withLedger(options, async ledger => {
+
+				if(options.speculos) {
```
Comment (3noch) [inline cli/cli.js:370]:
```suggestion
				if (options.speculos) {
```

Comment (3noch) [reply cli/cli.js:370]: Also no tabs.

File: cli-tests.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
+{ cli-app-avalanche, gecko, pkgs ? import <nixpkgs> {}}:
```
Comment (3noch) [inline cli-tests.nix:1]: Must pin this!

Comment (tomsmalley) [reply cli-tests.nix:1]: I think we are now pinning it in a bunch of places, it should get its own nix file.

File: cli/cli.js [hunk 7]
```diff
@@ -367,9 +367,9 @@ program
     if (address === undefined) {
       await withLedger(options, async ledger => {
 
-				if(options.speculos) {
-					flowAccept(ledger.transport)
-				}
+        if (options.speculos) {
```
Comment (3noch) [inline cli/cli.js:370]: We can fix this later but `speculos` should not imply *completely automated*.

**Comment by 3noch** [comment]
MR build fails due to CI bug. Commit build passes.

**Pull request merged**